editor/editor_node.cpp

```diff
@@ -6790,6 +6795,21 @@ bool EditorNode::call_build() {
 	return builds_successful;
 }
 
+// Check for available disk space on the target disk and warn the user if needed according to a set threshold.
+// The threshold (set in GiB) should be set to cover most use cases for the file being written.
+void EditorNode::check_disk_space(const String &p_target_path, float p_size_gib, const String &p_rationale) const {
+	Ref<DirAccess> dir = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+	String path = p_target_path.get_basename();
+	if (path.is_relative_path()) {
+		path = "res://" + path;
+	}
+	dir->open(path);
+	if (dir->get_space_left() < p_size_gib * Math::pow(1024.0, 3.0)) {
+		// Less than `p_size_gib` GiB available.
+		WARN_PRINT_ED(vformat(TTR("Current available space on target disk is low (%s)."), String::humanize_size(dir->get_space_left())) + " " + p_rationale);
```

Comment:
Could this include the actual disk that is being checked? e.g. on Windows the drive letter could be used. Not sure about Mac/Linux, or if there is an API to query such a thing already. If not, the path could be logged instead.

Same goes for the movie maker warning.